File: Reverse Integer/Leetcode_7.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        x = abs(x)
        res = 0
        while x:
            res = res*10 + x % 10
            x = x //10
        return 0 if (res < -2147483648 or res > 2147483647) else sign * res

File: Reverse Integer/Leetcode_7.cpp
#include <stdlib.h>
#include <limits.h>

class Solution {
public:
    int reverse(int x) {
        int res = 0;
        while (x!=0)
        {
            if (res > 0 and (res > INT_MAX/10 or res * 10 > INT_MAX - x % 10)) /*overflow */
                    return 0;
            if (res < 0 and (res < INT_MIN/10 or res * 10 < INT_MIN - x % 10)) /*underflow */
                    return 0;
            res = res * 10 + x %10; 
            x = x/10;
        }
        return res;
    }
};
